tests/tensor_base/test_astype.py

```diff
@@ -0,0 +1,33 @@
+import numpy as np
+from typing import Optional
+import hypothesis.extra.numpy as hnp
+
+from hypothesis import given
+import hypothesis.strategies as st
+
+from mygrad import Tensor
+from numpy.testing import assert_array_equal
+
+
+@given(
+    tensor=st.tuples(
+        hnp.arrays(
+            shape=hnp.array_shapes(), dtype=hnp.integer_dtypes() | hnp.floating_dtypes()
```

Comment:
does this cover uints or just signed ints?

Reply:
Yeah it is just signed ints, I can add unsigned